### ISSUE TITLE
[BUG] Jinja rendering on Debian 10 escapes safe strings

### ISSUE THREAD

**Issue by blarghmatey** [saltstack/salt#56833]
**Description**
On Debian 10 the rendering of JSON and YAML strings via Jinja is performing extra HTML escaping which breaks the installed applications.

**Setup**
As an example, the state definition [here](https://github.com/mitodl/elastic-stack-formula/blob/elastalert_venv/elastic-stack/elastalert/configure.sls#L15-L24) works fine on Debian 9, producing output of the form:

```yaml
foo:
  bar: "some string"
```

but the same contents on Debian 10 get rendered as:

```yaml
foo:
  bar: &#39;some string&#39;
```

**Steps to Reproduce the behavior**
Run the linked state with any settings definition which contains quotation marks on a Debian 10 host.

**Expected behavior**
The dictionary contents get rendered without any HTML escaping.

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
    Salt Version:
               Salt: 3000.1
     
    Dependency Versions:
               cffi: 1.14.0
           cherrypy: Not Installed
           dateutil: 2.7.3
          docker-py: Not Installed
              gitdb: Not Installed
          gitpython: Not Installed
             Jinja2: 2.10
            libgit2: Not Installed
           M2Crypto: Not Installed
               Mako: Not Installed
       msgpack-pure: Not Installed
     msgpack-python: 0.5.6
       mysql-python: Not Installed
          pycparser: 2.20
           pycrypto: 2.6.1
       pycryptodome: Not Installed
             pygit2: Not Installed
             Python: 3.7.3 (default, Dec 20 2019, 18:57:59)
       python-gnupg: Not Installed
             PyYAML: 3.13
              PyZMQ: 17.1.2
              smmap: Not Installed
            timelib: Not Installed
            Tornado: 4.5.3
                ZMQ: 4.3.1
     
    System Versions:
               dist: debian 10.3 
             locale: UTF-8
            machine: x86_64
            release: 4.19.0-8-cloud-amd64
             system: Linux
            version: debian 10.3
```
</details>

**Additional context**
In order to work around this problem we changed some of our state files to use the PyObjects renderer and avoid the trip through Jinja. That resolved the symptom but the underlying bug remains. An example can be found [here](https://github.com/mitodl/consul-formula/blob/master/consul/config.sls)

**Comment by blarghmatey** [comment]
@terminalmage let me know if you have any questions or need more detail. Thanks for the help!

**Comment by terminalmage** [comment]
No prob, I'll try to give this a look over the next few days.

**Comment by terminalmage** [comment]
I've traced this to a bug in the `indent` filter. If the value being indented is a `Markup` type (as is the case for any string passed through Salt's `yaml` filter), then once the lines are split they end up becoming strings, and when you concatenate a `Markup` object and a string, you get this:

```python
>>> from jinja2 import Markup
>>> Markup(u'foo:') + u'\n      bar: "baz"'
Markup(u'foo:\n      bar: &#34;baz&#34;')
```

The good news is that it is fixed upstream. The bad news is that it is only fixed in Jinja 2.11 and newer. The good news again though is that I can work around this by conditionally replacing the upstream `indent` filter with a version ported from 2.11 that contains the fix. So, anyone on a Jinja release < 2.11 will get the ported version, while anyone with Jinja >= 2.11 will get the upstream version of the `indent` filter.

I should be able to have a PR opened in the next couple days to fix this.

**Comment by terminalmage** [comment]
https://github.com/saltstack/salt/pull/56860
